sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientMetricsDiagnosticsHandler.java

```diff
@@ -9,26 +9,33 @@
 import com.azure.cosmos.CosmosDiagnosticsHandler;
 import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
 
+import java.lang.ref.WeakReference;
+
 import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
 
 public final class ClientMetricsDiagnosticsHandler implements CosmosDiagnosticsHandler {
     private static final ImplementationBridgeHelpers.CosmosClientTelemetryConfigHelper.CosmosClientTelemetryConfigAccessor clientTelemetryConfigAccessor =
         ImplementationBridgeHelpers.CosmosClientTelemetryConfigHelper.getCosmosClientTelemetryConfigAccessor();
 
-    private final CosmosAsyncClient client;
+    private final WeakReference<CosmosAsyncClient> clientRef;
 
     public ClientMetricsDiagnosticsHandler(CosmosAsyncClient client) {
 
         checkNotNull(client, "Argument 'client' must not be null.");
-        this.client = client;
+        this.clientRef =  new WeakReference<>(client);
     }
 
     @Override
     public void handleDiagnostics(CosmosDiagnosticsContext diagnosticsContext, Context traceContext) {
         checkNotNull(traceContext, "Argument 'traceContext' must not be null.");
 
+        CosmosAsyncClient clientSnapshot = this.clientRef.get();
+        if (clientSnapshot == null) {
```

Comment:
when can this happen?

Reply:
Only way to get here I can see are race conditions - remaining requests while client gets closed. Worst case we lose client-telemetry and/or metrics - nothing critical.